opslevel/common.go

```diff
@@ -32,6 +35,36 @@ import (
 
 // var DefaultPredicateDescription = "A condition that should be satisfied."
 
+const (
+	providerIssueUrl = "https://github.com/OpsLevel/terraform-provider-opslevel/issues"
+	providerFile     = "../tests/provider.tf"
+)
+
+type CommonClient struct {
+	client *opslevel.Client
+}
+
+// Configure sets up the OpsLevel client for datasources and resources
+func (d *CommonClient) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*opslevel.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *opslevel.Client, got: %T. Please report this issue to the provider developers at %s.", req.ProviderData, providerIssueUrl),
```

Comment:
not sure if this holds true for terraform errors but in golang error messages shouldn't have any capitals in text.  Maybe it doesn't matter here